src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java

```diff
@@ -234,21 +237,21 @@ public TextAssessmentResource(AuthorizationCheckService authCheckService, TextAs
     }
 
     /**
-     * Cancel an assessment of a given submission for the current user, i.e. delete the corresponding result / release the lock. Then the submission is available for assessment
+     * POST exercise/:exerciseId/submission/:submissionId/cancel-assessment : Cancel an assessment of a given submission for the current user, i.e. delete the corresponding result / release the lock. Then the submission is available for assessment
      * again.
      *
      * @param submissionId the id of the submission for which the current assessment should be canceled
      * @param exerciseId the exerciseId of the exercise for which the assessment gets canceled
      * @return 200 Ok response if canceling was successful, 403 Forbidden if current user is not the assessor of the submission
      */
-    @PutMapping("/exercise/{exerciseId}/submission/{submissionId}/cancel-assessment")
+    @PostMapping("/exercise/{exerciseId}/submission/{submissionId}/cancel-assessment")
```

Comment:
are you sure this should be a POST request? if you cancel the assessment, you basically delete something on the server

Reply:
I think it should be either DELETE but with the path "/exercise/{exerciseId}/submission/{submissionId}/assessment" or POST with the current path. basically every time the path ends in some sort of action (meaning verb) it should be POST, DELETE should only be done on a path that definitely identifies an object.

Reply:
Delete is not the same as cancel, I guess it can stay as is then, even if I think of POST as creating new data on the server, but maybe this understanding is wrong?